client/landing/stepper/declarative-flow/site-setup-flow.ts

```diff
@@ -627,13 +632,6 @@ const siteSetupFlow: Flow = {
 				case 'intent':
 					return exitFlow( `/home/${ siteId ?? siteSlug }` );
 
-				case 'goals':
-					// Skip to dashboard must have been pressed
-					setIntent( SiteIntent.Build );
-					return exitFlow( `/home/${ siteId ?? siteSlug }`, {
-						skipLaunchpad: true,
-					} );
-
```

Comment:
What's the goToNext => submit swap doing here? 

Was it intentional to no longer setIntent to the main build flow?

Reply:
`goNext` is marked as deprecated, and we use `submit` instead.

Despite being deprecated, the buttons in the step header component still only support using `goNext`, and so the "skip to dashboard" button in the header used to use `goNext`. In fact you can see that the code assumes that if `goNext` is called at all during the goals step then it must be a "skip to dashboard" action.

However now the "skip to dashboard" button has moved to the main body of the step and we can use `submit` navigation method instead.

The "skip" button in the header now "skips to the next step in the flow", not "skips to the dashboard"

Reply:
Ah, that makes sense, thanks.

> setIntent( SiteIntent.Build );

It wasn't clear, is that still being triggered somewhere for skips?

Reply:
Yes skipping will set `build`, and leaving the flow via DIFM or the Import link will set the expected intent.